scanpy/preprocessing/_normalization.py

```diff
@@ -14,14 +14,20 @@ def _normalize_data(X, counts, after=None, copy=False):
     X = X.copy() if copy else X
     if issubclass(X.dtype.type, (int, np.integer)):
         X = X.astype(np.float32)  # TODO: Check if float64 should be used
-    counts = np.asarray(counts)  # dask doesn't do medians
-    after = np.median(counts[counts>0], axis=0) if after is None else after
+    try:  # dask array
+        counts_greater_than_zero = counts[counts>0].compute_chunk_sizes()
```

Comment:
This is not perfectly efficient, we have to compute the chunk sizes here because they are unknown due to `counts[counts>0]`. There's probably still advantages here, compared with coercing the whole thing to a numpy array at the beginning.

See here for more details: https://docs.dask.org/en/latest/array-chunks.html
> `compute_chunk_sizes()` immediately performs computation and modifies the array in-place.